testutil/assertions/errors_test.go

```diff
@@ -0,0 +1,325 @@
+package assertions
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+)
+
+// errorContentsTestCase is a test case for the [Assert|Require]ErrorContents[f]? functions.
+type errorContentsTestCase struct {
+	name      string
+	theError  error
+	contains  []string
+	expOutput []string
+}
+
+// getErrorContentsTestCases returns all the tests cases for the [Assert|Require]ErrorContents[f]? functions.
+func getErrorContentsTestCases() []errorContentsTestCase {
+	return []errorContentsTestCase{
+		{
+			name:      "nil error nil contains",
+			theError:  nil,
+			contains:  nil,
+			expOutput: nil,
+		},
+		{
+			name:      "nil error one contains",
+			theError:  nil,
+			contains:  []string{"not in error"},
+			expOutput: []string{"An error is expected but got nil.", "Error was expected to contain: \"not in error\""},
+		},
+		{
+			name:      "nil error two contains",
+			theError:  nil,
+			contains:  []string{"missing", "fiddlesticks"},
+			expOutput: []string{"An error is expected but got nil.", "Error was expected to contain:\n\t\t\"missing\"\n\t\t\"fiddlesticks\""},
+		},
+		{
+			name:      "with error nil contains",
```

Comment:
Might be helpful to have a similar test, but where contains is []string{}.

Reply:
Added a couple test cases with an empty `contains`.